helix-rest/src/main/java/org/apache/helix/rest/server/service/InstanceServiceImpl.java

```diff
@@ -122,115 +122,166 @@ public InstanceInfo getInstanceInfo(String clusterId, String instanceName,
 
   /**
    * {@inheritDoc}
+   * Single instance stoppable check implementation is a special case of
+   * {@link #batchGetInstancesStoppableChecks(String, List, String)}
+   *
    * Step 1: Perform instance level Helix own health checks
    * Step 2: Perform instance level client side health checks
    * Step 3: Perform partition level (all partitions on the instance) client side health checks
-   * Note: if the check fails at one step, all the following steps won't be executed because the instance cannot be stopped
+   *
+   * Note: if the check fails at one step, the rest steps won't be executed because the instance cannot be stopped
    */
   @Override
-  public StoppableCheck getInstanceStoppableCheck(String clusterId, String instanceName,
+  public StoppableCheck getInstanceStoppableCheck(String clusterId, String instanceName, String jsonContent)
+      throws IOException {
+    return batchGetInstancesStoppableChecks(clusterId, ImmutableList.of(instanceName), jsonContent).get(instanceName);
+  }
+
+  @Override
+  public Map<String, StoppableCheck> batchGetInstancesStoppableChecks(String clusterId, List<String> instances,
       String jsonContent) throws IOException {
-    LOG.info("Perform instance level helix own health checks for {}/{}", clusterId, instanceName);
-    Map<String, Boolean> helixStoppableCheck = getInstanceHealthStatus(clusterId, instanceName,
-        InstanceService.HealthCheck.STOPPABLE_CHECK_LIST);
-
-    StoppableCheck result =
-        new StoppableCheck(helixStoppableCheck, StoppableCheck.Category.HELIX_OWN_CHECK);
-    if (!result.isStoppable()) {
-      return result;
+    Map<String, StoppableCheck> finalStoppableChecks = new HashMap<>();
+    Map<String, Future<StoppableCheck>> helixInstanceChecks = instances.stream()
```

Comment:
I think you need to add all instances keys to the finalStoppableChecks. Otherwise, it could return an empty map when we RESTConfig is not setup. 

The stoppable instances is also checked from final mapping. So either you change the top level result preparing or change the logic here.

Reply:
To be clearer, do you suggest to fill a default "true" stoppable check result for all instances first and let the rest steps update the map? This way we can guarantee any instance has a stoppable check result even in the case of failure. But it exposes another problem, if RESTConfig is not set and we just return dummy success, the client will treat the instances as stoppable. 

The other idea is to throw exceptions when abnormal things happen so the client won't get confused about the result.

Reply:
I have no preference for either way. Just want to make sure we deliver correct result.